test/Dialect/FIRRTL/imconstprop.mlir

```diff
@@ -542,3 +542,91 @@ firrtl.circuit "dntOutput" {
     firrtl.connect %b, %const : !firrtl.uint<3>, !firrtl.uint<3>
   }
 }
+
+// -----
+
+firrtl.circuit "VectorPropagation1" {
+  firrtl.module @VectorPropagation1(in %clock: !firrtl.clock, out %b: !firrtl.uint<1>) {
+    %c1_ui1 = firrtl.constant 1 : !firrtl.uint<1>
+    %r3 = firrtl.reg %clock  : !firrtl.vector<uint<1>, 2>
+    %0 = firrtl.subindex %r3[0] : !firrtl.vector<uint<1>, 2>
+    firrtl.connect %0, %c1_ui1 : !firrtl.uint<1>, !firrtl.uint<1>
+    %1 = firrtl.subindex %r3[1] : !firrtl.vector<uint<1>, 2>
+    firrtl.connect %1, %c1_ui1 : !firrtl.uint<1>, !firrtl.uint<1>
+    %2 = firrtl.xor %0, %1 : (!firrtl.uint<1>, !firrtl.uint<1>) -> !firrtl.uint<1>
+    firrtl.connect %b, %2 : !firrtl.uint<1>, !firrtl.uint<1>
+    // CHECK:      %c0_ui1 = firrtl.constant 0 : !firrtl.uint<1>
+    // CHECK-NEXT: firrtl.connect %b, %c0_ui1 : !firrtl.uint<1>, !firrtl.uint<1>
+  }
+}
+
+firrtl.circuit "VectorPropagation2" {
+  firrtl.module @VectorPropagation2(in %clock: !firrtl.clock, out %b1: !firrtl.uint<6>, out %b2: !firrtl.uint<6>, out %b3: !firrtl.uint<6>) {
```

Comment:
Most tests in here are split up (`// -----`).  It would be good to continue that convention.

Also, these tests look great.  For completeness, can you add a test that includes a bundle of vectors?

Reply:
Thanks, I will add tests with a bundle of vectors, and also asyncreset and nested bundles.